docs: clarify the parallelCodeSplitting and removeAvailableModules options (#10368)
web-infra-dev/rspack: website/docs/en/config/experiments.mdx

  experiments: {
    parallelCodeSplitting: true,
  },
  optimization: {
    removeAvailableModules: true,
  },
};
```

:::warning

> When `parallelCodeSplitting` is enabled, make sure to also enable `optimization.removeAvailableModules` (this is also enabled by default since version 1.3.0).
> This ensures consistency with the old code splitting algorithm, which enforced `removeAvailableModules` internally and ignored the `optimization.removeAvailableModules` configuration.

Reviewer: ```suggestion
When `parallelCodeSplitting` is enabled, ensure that 'optimization.removeAvailableModules' is also enabled (this has been enabled by default since version 1.3.0).

This maintains consistency with the previous code splitting algorithm, which enforced `removeAvailableModules` internally and ignored the `optimization.removeAvailableModules` configuration.
```
